portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -52,15 +60,22 @@ public void init() {
   }
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    int maxComments = Integer.parseInt(request.getParameter("max_comments"));
+    int commentsPerPage = Integer.parseInt(request.getParameter("comments_per_page")), pgNumber = Integer.parseInt(request.getParameter("pg_number")) - 1;
     Query comments_query = new Query("Comment").addSort("time_millis", SortDirection.DESCENDING);
-    List<Entity> results =  datastore.prepare(comments_query).asList(FetchOptions.Builder.withLimit(maxComments));
+    List<Entity> results =  datastore.prepare(comments_query).asList(FetchOptions.Builder.withLimit(commentsPerPage).offset(commentsPerPage * pgNumber));
```

Comment:
This line is quite long. You can break it with a newline before every "." in the call seqence.

Also, it looks like there are two spaces after the "=".